src/Sentry/Infrastructure/DiagnosticLogger.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Linq;
+using Sentry.Extensibility;
+
+namespace Sentry.Infrastructure
+{
+    /// <summary>
+    /// Base class for diagnostic loggers.
+    /// </summary>
+    public abstract class DiagnosticLogger : IDiagnosticLogger
+    {
+        private readonly SentryLevel _minimalLevel;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="DiagnosticLogger"/>.
+        /// </summary>
+        protected DiagnosticLogger(SentryLevel minimalLevel) => _minimalLevel = minimalLevel;
+
+        /// <summary>
+        /// Whether the logger is enabled to the defined level.
+        /// </summary>
+        public bool IsEnabled(SentryLevel level) => level >= _minimalLevel;
+
+        /// <summary>
+        /// Log message with level, exception and parameters.
+        /// </summary>
+        public void Log(SentryLevel logLevel, string message, Exception? exception = null, params object?[] args)
+        {
+            // Note, linefeed and newline chars are removed to guard against log injection attacks.
+            // See https://github.com/getsentry/sentry-dotnet/security/code-scanning/5
+
+            var formattedMessage = new string(string.Format(message, args)
```

Comment:
for a log message of 
```
The log
message
```

wont this result in `The logmessage` ? Is that what we want?

Reply:
Do you think it would be better to do something like replacing `"\n"` with `"\\n"`, etc.?

Reply:
i was thinking replacing with a space would be good enough

Reply:
Done. Thanks.